app/models/table.rb

```diff
@@ -0,0 +1,17 @@
+class Table < ApplicationRecord
+
+  belongs_to :branch
+  has_and_belongs_to_many :reservations
+
+  def vacant?
+    !reservations.active.exists?
+  end
+
+  def vacant_for?(new_reservation)
+    reservations.each do |reservation|
```

Comment:
look for iterators in ruby that return boolean

Reply:
Fixed.